cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/data/SidecarTableSizeProvider.java

```diff
@@ -0,0 +1,151 @@
+    {
+        try
+        {
+            RingResponse ringResponse = ringFuture.toCompletableFuture().get();
+            return tableSizeInBytesInternal(ringResponse, keyspace, table, datacenter);
+        }
+        catch (Exception ex)
+        {
+            throw new RuntimeException(String.format("Error occurred while determining the table size for table '%s.%s'",
+                                                     keyspace, table), ex);
+        }
+    }
+
+    private long tableSizeInBytesInternal(RingResponse ringResponse, String keyspace, String table, String datacenter)
+    {
+        LongAdder tableSize = new LongAdder();
+        LongAdder success = new LongAdder();
+        List<RingEntry> instances = ringResponse.stream()
+                                                .filter(ringEntry -> datacenter == null || datacenter.equals(ringEntry.datacenter()))
+                                                .collect(Collectors.toUnmodifiableList());
+        int instancesSize = instances.size();
+        CountDownLatch latch = new CountDownLatch(instancesSize);
+
+        for (RingEntry ringEntry : instances)
+        {
+            SidecarInstance sidecarInstance = new SidecarInstanceImpl(ringEntry.fqdn(), sidecarClientConfig.effectivePort());
+            CompletableFuture<TableStatsResponse> tableStatsResponseCompletableFuture = sidecarClient.tableStats(sidecarInstance, keyspace, table);
+            tableStatsResponseCompletableFuture
+            .whenComplete((response, throwable) -> {
+                try
+                {
+                    if (throwable == null)
+                    {
+                        tableSize.add(response.totalDiskSpaceUsedBytes());
+                        success.increment();
+                    }
+                    else
+                    {
+                        LOGGER.warn("Failed to retrieve table statistics for keyspace {} table {} on instance {}",
+                                    keyspace, table, sidecarInstance, throwable);
+                    }
+                }
+                finally
+                {
+                    latch.countDown();
+                }
+            });
+        }
+
+        // wait until all the calls to table stats have completed
+        await(latch);
+
+        long successCount = success.sum();
+        long tableSizeSum = tableSize.sum();
+        if (successCount == 0)
+        {
+            throw new RuntimeException(String.format("Unable to determine the table size for table '%s.%s'. 0/%d instances available.",
+                                                     keyspace, table, instancesSize));
+        }
+        else if (successCount < instancesSize)
+        {
+            // normalize table size to the number of instances
+            LOGGER.debug("{}/{} instances were used to determine the table size {} for table {}.{}",
+                         successCount, instancesSize, tableSizeSum, keyspace, table);
```

Comment:
How about logging at `info` level? It is logged only once per job and the information is seemingly useful for debugging.